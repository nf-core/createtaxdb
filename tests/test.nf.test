nextflow_pipeline {

    name "Test pipeline: NFCORE_CREATETAXDB"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_createtaxdb"
    tag "test"

    test("test_profile") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                        file("$outputDir/bracken/database/database100mers.kmer_distrib").name,
                        file("$outputDir/bracken/database/database100mers.kraken").name,
                        file("$outputDir/bracken/database/database.kraken").name,
                        path("$outputDir/centrifuge/"),
                        path("$outputDir/diamond/database.dmnd"),
                        path("$outputDir/kaiju/database.fmi"),
                        path("$outputDir/kraken2/database/hash.k2d"),
                        file("$outputDir/kraken2/database/opts.k2d").name,
                        path("$outputDir/kraken2/database/taxo.k2d"),
                        path("$outputDir/krakenuniq/database/database-build.log").readLines().last().contains('database.idx'),
                        file("$outputDir/krakenuniq/database/database.idx").name,
                        file("$outputDir/krakenuniq/database/database.kdb"),
                        file("$outputDir/krakenuniq/database/taxDB"),
                        path("$outputDir/malt/malt-build.log").readLines().last().contains('Peak memory'),
                        path("$outputDir/malt/malt_index/index0.idx"),
                        path("$outputDir/malt/malt_index/ref.db"),
                        path("$outputDir/malt/malt_index/ref.idx"),
                        path("$outputDir/malt/malt_index/ref.inf"),
                        path("$outputDir/malt/malt_index/taxonomy.idx"),
                        path("$outputDir/malt/malt_index/taxonomy.map"),
                        path("$outputDir/malt/malt_index/taxonomy.tre")
                    ).match()
                },
                { assert new File("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
                { assert path("$outputDir/malt/malt_index/table0.db").exists() },
                { assert path("$outputDir/malt/malt_index/table0.idx").exists() },
            )
        }
    }
}
