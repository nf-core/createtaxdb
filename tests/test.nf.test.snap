{
    "test_profile": {
        "content": [
            "database100mers.kmer_distrib",
            "database100mers.kraken",
            "database.kraken",
            [
                "database.1.cf:md5,1481615ab90b5573f6d9e57f97890178",
                "database.2.cf:md5,d50fa66e215e80284314ff6521dcd4a4",
                "database.3.cf:md5,beafa92166ba461f9bda1dac0b640f45",
                "database.4.cf:md5,2902ec5df0db6da41a91b40d2f46b30d"
            ],
            "database.dmnd:md5,b2ea49ef5490c526e2c56cae19bcb462",
            "database.fmi:md5,54fd89f5e4eab61af30175e8aa389598",
            "hash.k2d:md5,941118164b4bcc010593f7a7c7b30029",
            "opts.k2d",
            "taxo.k2d:md5,cd8170a8c5a1b763a9ac1ffa2107cc88",
            true,
            "database.idx",
            "database.kdb:md5,a24fce43bedbc6c420f6e36d10c112a3",
            "taxDB:md5,1aed1afa948daffc236deba1c5d635db",
            true,
            "index0.idx:md5,876139dc930e68992cd2625e08bba48a",
            "ref.db:md5,377073f58a9f9b85acca59fcf21744a9",
            "ref.idx:md5,676393d0f4826dac3f47aa5290632570",
            "ref.inf:md5,446275f4f879e2b115e983db7c9ced18",
            "taxonomy.idx:md5,1e2e8fdc703a6d2707e7cbefd2b6d93f",
            "taxonomy.map:md5,5bb3f2192e925bca2e61e4b54f1671e0",
            "taxonomy.tre:md5,f76fb2d5aa9b0d637234d48175841e0e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-08T16:33:06.699148849"
    }
}